### PR TITLE
Make PiP motion smoother and react to window resizes correctly

### PR DESCRIPTION
**Pull request by robintown**
This uses `transform` rather than `inset` to make the motion of the PiP smoother, and sets the style on the DOM node directly rather than going through React for better performance.

Closes https://github.com/vector-im/element-web/issues/22292

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Make PiP motion smoother and react to window resizes correctly ([\#8747](https://github.com/matrix-org/matrix-react-sdk/pull/8747)). Fixes vector-im/element-web#22292.

### CODE REVIEW

File: src/components/views/elements/PersistedElement.tsx [hunk 1]
```diff
@@ -177,8 +182,9 @@ export default class PersistedElement extends React.Component<IProps> {
         Object.assign(child.style, {
             zIndex: isNullOrUndefined(this.props.zIndex) ? 9 : this.props.zIndex,
             position: 'absolute',
-            top: parentRect.top + 'px',
-            left: parentRect.left + 'px',
+            top: '0',
+            left: '0',
```
Comment (SimonBrandner) [inline src/components/views/elements/PersistedElement.tsx:186]: Do we need to assign 0s here?

File: src/components/views/voip/PictureInPictureDragger.tsx [hunk 2]
```diff
@@ -43,69 +43,67 @@ interface IProps {
...
+
+        this.props.onMove?.();
+    };
+
+    private setStyle = () => {
+        if (this.callViewWrapper.current) {
+            // Set the element's style directly, bypassing React for efficiency
+            this.callViewWrapper.current.style.transform =
+                `translateX(${this.translationX}px) translateY(${this.translationY}px)`;
+        }
```
Comment (SimonBrandner) [inline src/components/views/voip/PictureInPictureDragger.tsx:106]:
```
if (!this.callViewWrapper.current) return
```

might be nicer but it also might be a personal preference

**Review by SimonBrandner — APPROVED** [review]
Otherwise looks sane

File: src/components/views/elements/PersistedElement.tsx (hunk 1 above)
Comment (robintown) [reply src/components/views/elements/PersistedElement.tsx:186]: Yep, otherwise the element gets placed somewhere offscreen

Comment (SimonBrandner) [reply src/components/views/elements/PersistedElement.tsx:186]: Since it's always the same could it be set in CSS? Am I missing something?

**Pull request merged**